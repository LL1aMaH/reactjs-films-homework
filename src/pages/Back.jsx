import React from 'react';

import { connect } from 'react-redux';
import Search from '../components/search';
import Button from '../components/button';
import Name from '../components/name';
import store from '../redux/store';
import searchName from '../redux/actions/actions';
import './Back.scss';

import { nameButton1, nameButton2, dataFilm } from '../config.data'; // data for test

const handleClick = (text) => {
  console.log(text[text.length - 1]);
};

const onInput = () => {
  store.dispatch(searchName(document.querySelector('.js-search').value));
  store.subscribe(() => console.info(store.getState()));
};

const Back = () => (
  <div className="background">
    <div className="search-block">
      <Search onInput={onInput} />
    </div>
    <div className="bottom-block">
      <div className="name-block">
        <Name dataFilm={dataFilm} />
      </div>
      <div className="bottom-button-block">
        <Button text={nameButton1} onClick={handleClick} />
        <Button text={nameButton2} onClick={handleClick} />
      </div>
    </div>
  </div>
);

export default connect(null)(Back);
