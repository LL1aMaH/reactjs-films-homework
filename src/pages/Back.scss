.background {
  background-image: url('../public/cover-image.jpg');
  background-size: cover;
  height: 500px;
  min-width: 100%;
  margin: 0 auto;
  display: flex;
  padding: 0 0 10px 0;
  justify-content: space-between;
  flex-direction: column;
}

.bottom-block {
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.bottom-button-block {
  align-self: flex-end;
  display: flex;
  flex-direction: row;
}

.search {
  display: flex;
  margin-top: 15px;
  margin-right: 20px;
}

.button {
  margin-right: 20px;
}

.search-block {
  width: 100%;
  display: flex;
  justify-content: flex-end;
}

.name-block {
  margin-left: 20px;
}

.posters{
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: flex-start;
  flex-wrap: wrap;
  align-content: space-around;

}

.poster{
  display: inline;
  padding: 5% 5% 0 5%;
}
.poster img{
  max-height: 300px;
}