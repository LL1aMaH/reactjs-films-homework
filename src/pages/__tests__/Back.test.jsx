import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Back from '../Back';

describe('<Back />', () => {
  it('<Back />', () => {
    const wrapper = shallow(<Back />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
