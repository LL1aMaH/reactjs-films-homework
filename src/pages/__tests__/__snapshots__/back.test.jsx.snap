// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Back /> <Back /> 1`] = `
<div
  className="background"
>
  <div
    className="search-block"
  >
    <Search />
  </div>
  <div
    className="bottom-block"
  >
    <div
      className="name-block"
    >
      <Name
        dataFilm={
          Object {
            "genre": "action",
            "name": "Gentleman",
            "stars": 10,
            "time": "113 min",
          }
        }
      />
    </div>
    <div
      className="bottom-button-block"
    >
      <Button
        onClick={[Function]}
        text="Button1"
      />
      <Button
        onClick={[Function]}
        text="Button2"
      />
    </div>
  </div>
</div>
`;
